src/python/espressomd/visualizationOpenGL.pyx

```diff
@@ -204,18 +326,18 @@ def drawSystemParticles(self):
             ptype = self.particles['types'][pid]
             ext_f = self.particles['ext_forces'][pid]
 
-            # Size: Lennard Jones Sigma,
-            if self.specs['particle_sizes'] == 'auto':
-                lj_sig = self.system.non_bonded_inter[ptype, ptype].lennard_jones.get_params()[
-                    'sigma']
-                radius = lj_sig / 2.0
-                if radius == 0:
-                    radius = self.sizeByType(ptype)
-
-            elif self.specs['particle_sizes'] == 'type':
-                radius = self.sizeByType(ptype)
+            radius = 0.5
+            try:
+                if callable(self.specs['particle_sizes']):
+                    radius = self.specs['particle_sizes'](ptype)
+                elif isinstance(self.specs['particle_sizes'], (list,tuple,np.ndarray)):
```

Comment:
I see your point, but the try..except now prevents 
-corrupted lambda functions
-errors from `self.system.non_bonded_inter[ptype, ptype].lennard_jones.get_params()`
Duck-typing here would mean to check for __iter__ and types of list members?
I think it is pretty hard to catch all (user-induced) irregular parameters. e.g. the mayavi approach fails if the uses types 'autos', which would end up in radius = 'a' for type 0..